src/Icicle/Pipeline.hs

```diff
@@ -81,7 +93,7 @@ annotOfError e
     CompileErrorProgram _
      -> Nothing
 
-instance Pretty a => Pretty (CompileError a) where
+instance (IsString b, Ord b, Pretty a, Pretty b, Show a, Show b, Show c) => Pretty (CompileError a b c) where
```

Comment:
It contains CheckError, which has suggestions, which calls prettyFunFromStrings.